server/users.ts

```diff
@@ -159,7 +159,12 @@ function findUsers(userids: ID[], ips: string[], options: {forPunishment?: boole
 		for (const myIp of ips) {
 			if (user.ips.includes(myIp) || (
 				(myIp.includes('*') || myIp.includes('-')) &&
-				user.ips.map(IPTools.ipToNumber).some(ip => {
+				user.ips.map(ip => {
+					const ipNumber = IPTools.ipToNumber(ip);
+					// this should never happen, but if a user object has an invalid IP address, that's pretty concerning
+					if (ipNumber === null) throw new Error(`Invalid IP address on ${user.id}: '${ip}'`);
+					return ipNumber;
+				}).some(ip => {
```

Comment:
I believe this is known to happen sometimes due to a SockJS bug, and I really don't want to take down the entire server in that case.

Reply:
We already handle that case in

Reply:
I think a lot of these checks can be replaced with having functions like `checkPattern` or `stringToRange` accept `null`.